spring-integration-core/src/main/java/org/springframework/integration/store/AbstractKeyValueMessageStore.java

```diff
@@ -156,6 +157,24 @@ public MessageGroup removeMessageFromGroup(Object groupId, Message<?> messageToR
 
 
 	@Override
+	public void removeMessagesFromGroup(Object groupId, Collection<Message<?>> messages) {
+		Assert.notNull(groupId, "'groupId' must not be null");
+		Assert.notNull(messages, "'messages' must not be null");
+
+		Object mgm = this.doRetrieve(MESSAGE_GROUP_KEY_PREFIX + groupId);
+		if (mgm != null) {
+			Assert.isInstanceOf(MessageGroupMetadata.class, mgm);
+			MessageGroupMetadata messageGroupMetadata = (MessageGroupMetadata) mgm;
+			messageGroupMetadata.setLastModified(System.currentTimeMillis());
+			for (Message<?> messageToRemove : messages) {
+				this.removeMessage(messageToRemove.getHeaders().getId());
+			}
+			MessageGroup messageGroup = this.getSimpleMessageGroup(this.getMessageGroup(groupId));
+			this.doStore(MESSAGE_GROUP_KEY_PREFIX + groupId, new MessageGroupMetadata(messageGroup));
```

Comment:
This two lines don't do anything proper. From other side you don't update `MessageGroupMetadata` in the persistence store.
For that looks like:
````java
Assert.isInstanceOf(MessageGroupMetadata.class, mgm);
MessageGroupMetadata messageGroupMetadata = (MessageGroupMetadata) mgm;
...
UUID messageId = messageToRemove.getHeaders().getId();
...
messageGroupMetadata.remove(messageId);
....
doStore(MESSAGE_GROUP_KEY_PREFIX + groupId, messageGroupMetadata);
````